java/java.source.base/src/org/netbeans/modules/java/source/save/Reformatter.java

```diff
@@ -2659,11 +2750,17 @@ public Boolean visitCase(CaseTree node, Void p) {
             } else {
                 accept(DEFAULT);
             }
+            List<? extends StatementTree> statements = node.getStatements();
+            if(statements != null)
             accept(COLON);
+            else if (node instanceof JCTree.JCCase) {
+                statements = ((JCTree.JCCase) node).stats;
```

Comment:
Sorry, but it is difficult to say if this approach works without testing. In some cases "stats" may contain synthesized trees, for things like:
```
int x = switch (...) {
    case 0 -> 0;
    default -> -1;
}
```

"stats" contains a synthetic Break statement, which refers to "0" and "-1".

I guess I'd suggest to try without referring to the field, but rather use TreeShims.getBody(), although I admit I am not sure where that would lead.